src/components/Project/ImageAnalysis/ImageViewer.vue

```diff
@@ -105,96 +114,65 @@ function leafletSetup(){
     drawText: false,
     drawRectangle: false,
     editMode: true,
-    dragMode: true,
+    dragMode: false,
     cutPolygon: false,
     rotateMode: false,
     removalMode: true
   })
 
-  const zoomedInThreshold = 1
-
-  // Disable dragging until zoomed in
-  image.on('zoomend', () => {
-    if (image.getZoom() >= zoomedInThreshold) {
-      image.dragging.enable()
-    } else {
-      image.dragging.disable()
-    }
-  })
-
-  // Finish line after 2 points (default is multiple points for a line)
-  image.on('pm:drawstart', ({ workingLayer }) => {
+  imageMap.on('pm:drawstart', ({ workingLayer }) => {
     // Remove last drawn line when starting new one
-    if (lastDrawnLine && image.hasLayer(lastDrawnLine)) {
-      image.removeLayer(lastDrawnLine)
+    if (lineLayer && imageMap.hasLayer(lineLayer)) {
+      imageMap.removeLayer(lineLayer)
     }
-    let pointsCount = 0
+    // Limit line to 2 points
     workingLayer.on('pm:vertexadded', () => {
-      pointsCount++
-      if (pointsCount === 2) {
-        image.pm.Draw.Line._finishShape()
+      if (imageMap.pm.Draw.Line._markers.length === 2) {
+        imageMap.pm.Draw.Line._finishShape()
       }
     })
   })
 
   // Get coordinates of the line
-  image.on('pm:create', (e) => {
+  imageMap.on('pm:create', (e) => {
     // Save last drawn line
-    lastDrawnLine = e.layer
-    e.layer.on('pm:edit', handleEdit)
-    requestLineProfile(e.layer.getLatLngs())
+    lineLayer = e.layer
+    lineLayer.on('pm:edit', handleEdit)
+    requestLineProfile(lineLayer.getLatLngs())
   })
 }
 
-// Checks if the point is within the bounds of the image
-function isWithinBounds(point) {
-  const latWithinBounds = point.lat >= 0 && point.lat <= imageBounds[1][0]
-  const lngWithinBounds = point.lng >= 0 && point.lng <= imageBounds[1][1]
-  return latWithinBounds && lngWithinBounds
+// Recalculates image size and position after the container's dimensions have changed
+function resizeMap(){
+  const resizeObserver = new ResizeObserver(() => {
+    imageMap.invalidateSize()
+  })
+  if(imageContainer.value){
+    resizeObserver.observe(imageContainer.value)
+  }
 }
 
 // Adjusts the point to be within the bounds of the image
 function adjustToBounds(point) {
   const lat = Math.max(0, Math.min(imageBounds[1][0], point.lat))
   const lng = Math.max(0, Math.min(imageBounds[1][1], point.lng))
-  // Returns a new point with adjusted coordinates if the point is outside the bounds
   return L.latLng(lat, lng)
 }
 
-// Callback function for when a line is edited and checks if it's within bounds
```

Comment:
Wasn't necessary to delete the old line and draw a new on on edit. Still bounds the line by the map bounds and instead just updates the line with the new Lat Lngs and requests a line profile